### ISSUE TITLE
samples/modules/chre/sample.modules.chre failing for esp32s3_devkitm since 4b5331ba45

### ISSUE THREAD

**Issue by fabiobaltieri** [zephyrproject-rtos/zephyr#63986]
**Describe the bug**
Hi, `samples/modules/chre/sample.modules.chre` started to fail linking in CI since 4b5331ba45 (https://github.com/zephyrproject-rtos/zephyr/pull/63817):

```
$ west build -p -b esp32s3_devkitm -T samples/modules/chre/sample.modules.chre
...
/opt/zephyr-sdk-0.16.3/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd:zephyr/linker_zephyr_pre0.cmd:293 cannot move location counter backwards (from 000000003fc88010 to 000000003fc88000)
```

Can't quite figure what's wrong with it, seems like it does not like the `KEEP(*(".unstable_id"));` being there but I can't quite tell why.

@sylvioalves is this something you could look into?

It's quite an edge case so I would not mark it as release blocker, but it is broken in CI so it it should have at least be looked into before the release.

**To Reproduce**
`west build -p -b esp32s3_devkitm -T samples/modules/chre/sample.modules.chre`

**Expected behavior**
Builds

**Impact**
CI breakage

**Environment (please complete the following information):**
Linux, Zephyr SDK 0.16.3, 90b9809c6f

**Comment by fabiobaltieri** [comment]
cc @jhedberg

**Comment by sylvioalves** [comment]
@fabiobaltieri, I can't even build it. Am I missing anything?
```c
/home/sylvio/espressif/zephyr/zephyr/samples/modules/chre/prj.conf:1: warning: attempt to assign the value 'y' to the undefined symbol CHRE

/home/sylvio/espressif/zephyr/zephyr/samples/modules/chre/prj.conf:4: warning: attempt to assign the value 'y' to the undefined symbol CHRE_LOG_LEVEL_DBG
Parsing /home/sylvio/espressif/zephyr/zephyr/Kconfig
Loaded configuration '/home/sylvio/espressif/zephyr/zephyr/boards/xtensa/esp32_devkitc_wroom/esp32_devkitc_wroom_defconfig'
Merged configuration '/home/sylvio/espressif/zephyr/zephyr/samples/modules/chre/prj.conf'

error: Aborting due to Kconfig warnings

CMake Error at /home/sylvio/espressif/zephyr/zephyr/cmake/modules/kconfig.cmake:355 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/sylvio/espressif/zephyr/zephyr/cmake/modules/zephyr_default.cmake:129 (include)
  /home/sylvio/espressif/zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/sylvio/espressif/zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:6 (find_package)
```

**Comment by fabiobaltieri** [comment]
Oh, sorry, yes:
```
west config manifest.group-filter -- +optional
west update
```

chre has been moved to the optional manifest, the CI fetches those as well.

**Comment by sylvioalves** [comment]
cc @yperess
@fabiobaltieri I was checking chre module and found this:  
```c
zephyr_linker_sources(SECTIONS linker_chre.ld)
```
https://github.com/zephyrproject-rtos/chre/blob/zephyr/platform/zephyr/CMakeLists.txt#L59

This linker file creates ram sections:  
```c
SECTION_DATA_PROLOGUE(_CHRE_SECTION,,)
{
	KEEP(*(".unstable_id"));
} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
```

Should this be changed to a different linking macro:  
`zephyr_linker_sources(RAM_SECTIONS linker_chre.ld)`

Sylvio

**Comment by fabiobaltieri** [comment]
Looks like the data in that section does not really need to be linked in RAM, changing to `GROUP_ROM_LINK_IN` fixes the issue

https://github.com/zephyrproject-rtos/chre/pull/9/files
https://github.com/zephyrproject-rtos/zephyr/pull/64019

**Comment by fabiobaltieri** [comment]
Fixed by #64019
